### DISCUSSION TITLE
Confusion with CURLINFO_OS_ERRNO and its example usage in libcurl documentation

### DISCUSSION THREAD

**Discussion by farazrbx** [curl/curl#13557] [Q&A] [answered]
I'm having some trouble understanding how to use `CURLINFO_OS_ERRNO` in my app.
Here's the example code snippet that I've copied from [CURLINFO_OS_ERRNO](https://curl.se/libcurl/c/CURLINFO_OS_ERRNO.html)

```c
long error;
res = curl_easy_getinfo(curl, CURLINFO_OS_ERRNO, &error);
if (res && error) {
    printf("Errno: %ld\n", error);
}
```

However, the documentation also states that `curl_easy_getinfo` returns `CURLE_OK` if the option is supported. But `CURLE_OK` is defined as 0. So is this a mistake in the example code? Do we need to change the if statement to `res == CURLE_OK && error`?

**Comment by jay** [comment]
> res = curl_easy_getinfo(curl, CURLINFO_OS_ERRNO, &error);
> if (res && error) {

That's a bug, I will take a look at the CURLINFO docs

> Do we need to change the if statement to `res == CURLE_OK && error`?

yes

**Reply by jay** [reply, marked as answer]
it's a bug, i found a few, please take further discussion to #13559
